lib/main.coffee

```diff
@@ -62,9 +71,11 @@ module.exports =
       lint: (activeEditor) =>
         file = activeEditor.getPath()
         return helpers.tempFile path.basename(file), activeEditor.getText(), (tmpFilename) =>
-          projDir = @getProjDir(file)
-          cwd = projDir
-          pythonPath = @pythonPath.replace(/%p/g, projDir)
+          # default project dir to file directory if path cannot be determined
+          projDir = @getProjDir(file) or path.dirname(file)
+          cwd = @cwd.replace('%f', path.dirname(file)).replace('%p', projDir)
+          executable = @executable.replace('%p', projDir)
+          pythonPath = @pythonPath.replace('%p', projDir)
```

Comment:
Please switch this back to using global replacement for `%p`, since we may want to add multiple subdirectories of the project dir to `PYTHONPATH`

```js
pythonPath = @pythonPath.replace(/%p/g, projDir)
```